yt/yt/server/http_proxy/component_discovery.cpp

```diff
@@ -149,15 +171,22 @@ std::vector<TClusterComponentInstance> TComponentDiscoverer::ListProxies(ECluste
             YT_ABORT();
     }
 
+    options.MaxSize = DefaultCypressMaxSize;
+
     auto nodeYson = WaitFor(Client_->GetNode(GetCypressDirectory(component), options))
         .ValueOrThrow();
-    auto addressToNode = ConvertTo<THashMap<TString, IMapNodePtr>>(nodeYson);
+    auto nodeMap = ConvertTo<IMapNodePtr>(nodeYson);
+
+    if (nodeMap->Attributes().Get("incomplete", false)) {
```

Comment:
Let extract this check to separate method `ValueResultCompleteness` and avoid copypaste